ui2/src/pages/login/QuickSetup.tsx

```diff
@@ -0,0 +1,560 @@
+import React, { useEffect, useState } from 'react';
+
+import { Form, Button, Icon, Loader, Table } from 'semantic-ui-react';
+
+import DamlLedger, { useLedger, useStreamQueries } from '@daml/react';
+import { DablPartiesInput, PartyDetails } from '@daml/hub-react';
+import { useWellKnownParties } from '@daml/hub-react/lib';
+
+import { useUserDispatch } from '../../context/UserContext';
+import { useHistory } from 'react-router-dom';
+
+import {
+  DeploymentMode,
+  deploymentMode,
+  httpBaseUrl,
+  wsBaseUrl,
+  ledgerId,
+  publicParty,
+} from '../../config';
+
+import Credentials, { computeCredentials } from '../../Credentials';
+import { retrieveParties, storeParties } from '../../Parties';
+
+import { halfSecondPromise } from '../page/utils';
+
+import { Service as OperatorService } from '@daml.js/da-marketplace/lib/Marketplace/Operator/Service';
+import {
+  Offer as CustodianOffer,
+  Role as CustodianRole,
+} from '@daml.js/da-marketplace/lib/Marketplace/Custody/Role';
+import {
+  Offer as DistributorOffer,
+  Role as DistributorRole,
+} from '@daml.js/da-marketplace/lib/Marketplace/Distribution/Role';
+import {
+  Offer as SettlementOffer,
+  Service as SettlementService,
+} from '@daml.js/da-marketplace/lib/Marketplace/Settlement/Service';
+import {
+  Offer as ExchangeOffer,
+  Role as ExchangeRole,
+} from '@daml.js/da-marketplace/lib/Marketplace/Trading/Role';
+import {
+  Offer as MatchingOffer,
+  Service as MarchingService,
+} from '@daml.js/da-marketplace/lib/Marketplace/Trading/Matching/Service';
+
+import { CreateEvent } from '@daml/ledger';
+
+type Offer = CustodianOffer | DistributorOffer | SettlementOffer | ExchangeOffer | MatchingOffer;
+
+enum serviceOptionsEnum {
+  CUSTODY = 'Custody',
+  TRADING = 'Trading',
+  MATCHING = 'Matching',
+  SETTLEMENT = 'Settlement',
+  LISTING = 'Listing',
+}
```

Comment:
I defined these data types in my recent PR, would be good to switch over! 

`ServiceOffer` imported from `context/ServicesContext.tsx` and 

`ServiceKind` imported from the same file. It is also a service string enum 

You'll have to add some additional services (e.g., Matching) in those though